journalbeat/reader/journal.go

```diff
@@ -288,8 +288,14 @@ func (r *Reader) convertNamedField(fc fieldConversion, value string) interface{}
 	if fc.isInteger {
 		v, err := strconv.ParseInt(value, 10, 64)
 		if err != nil {
-			r.logger.Debugf("Failed to convert field: %s \"%v\" to int: %v", fc.name, value, err)
-			return value
+			// Failed to convert to integer, try to strip ',\w*' from the
+			// end of of the value and try again.
+			s := strings.Split(value, ",")
```

Comment:
Any reason not to try and run this split always before the `strconv.ParseInt(...)`, so basically right inside the `if fc.isInteger` block? Are you doing this lazily as a performance optimization?

Reply:
Yes doing it lazily for performance because even on my system I could not get this case to occur. I don't know ifs its an older systemd or what that appends the username to the PID. I thought it would be best to only do the split in the case that converting to an integer failed.

Reply:
Cool, makes sense then. Maybe add a comment here about that?

Reply:
Okay will add a comment.